src/test/java/com/users/service/UserServiceTest.java

```diff
@@ -0,0 +1,244 @@
+package com.users.service;
+
+import com.users.criteria.UserCriteriaEasy;
+import com.users.dto.UserDto;
+import com.users.entities.User;
+import com.users.exceptions.EntityAlreadyExistsException;
+import com.users.exceptions.EntityNotFoundException;
+import com.users.mappers.UserMapper;
+import com.users.repository.UserRepository;
+import com.users.utility.UserUtilityTest;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
```

Comment:
de ce ai pus mock aici si injectMocks la urmatorul?

Reply:
ca sa injectez Mockurile respective in serviciul meu, mock urile sa fie initializate cu ajutorul adnotarii puse pe clasa si injectate in serviciul meu

Reply:
cum adica sa injectezi mockruile?

Reply:
din ce inteleg eu si vad si pe net, in clasa mea Service ma folosesc de repository si de mapper si asta fac si in clasa de teste unitare. Am nevoie de repository si de mapper, care sunt varibile in clasa Service si din acest motiv sunt adnotate cu @Mock, iar userService cu InjectMocks, pentru a injecta cele 2 variabile in clasa mea de service si a le putea folosi la testare